### ISSUE TITLE
structcheck: false positive

### ISSUE THREAD

**Issue by gucio321** [golangci/golangci-lint#2054]
- [x] Yes, I'm using a binary release within 2 latest major releases. Only such installations are supported.
- [x] Yes, I've searched similar issues on GitHub and didn't find any.
- [x] Yes, I've included all information below (version, config, etc).
- [ ] Yes, I've tried with the standalone linter if available. (https://golangci-lint.run/usage/linters/)

<details><summary>Description of the problem</summary>

Hi,
I noticed a very strange begavior of `stuctcheck` linter, which report false-positive on... unexpored fields of structures in some specific situations

</details>

<details><summary>Version of golangci-lint</summary>

```console
$ golangci-lint --version
golangci-lint has version 1.40.1 built from 625445b1 on 2021-05-14T11:44:45Z
```

</details>

<details><summary>Config file</summary>

```console
$ cat .golangci.yml
linters:
  disable-all: true
  enable:
    - structcheck
```

</details>

<details><summary>Go environment</summary>

```console
$ go version && go env
go version go1.16.4 linux/amd64
GO111MODULE=""
GOARCH="amd64"
GOBIN=""
GOCACHE="/home/username/.cache/go-build"
GOENV="/home/username/.config/go/env"
GOEXE=""
GOFLAGS=""
GOHOSTARCH="amd64"
GOHOSTOS="linux"
GOINSECURE=""
GOMODCACHE="/home/username/go/pkg/mod"
GONOPROXY=""
GONOSUMDB=""
GOOS="linux"
GOPATH="/home/username/go"
GOPRIVATE=""
GOPROXY="https://proxy.golang.org,direct"
GOROOT="/usr/local/go"
GOSUMDB="sum.golang.org"
GOTMPDIR=""
GOTOOLDIR="/usr/local/go/pkg/tool/linux_amd64"
GOVCS=""
GOVERSION="go1.16.4"
GCCGO="gccgo"
AR="ar"
CC="gcc"
CXX="g++"
CGO_ENABLED="1"
GOMOD="/dev/null"
CGO_CFLAGS="-g -O2"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-g -O2"
CGO_FFLAGS="-g -O2"
CGO_LDFLAGS="-g -O2"
PKG_CONFIG="pkg-config"
GOGCCFLAGS="-fPIC -m64 -pthread -fmessage-length=0 -fdebug-prefix-map=/tmp/go-build418452944=/tmp/go-build -gno-record-gcc-switches"
```

</details>

<details><summary>Code example or link to a public repository</summary>

sadly, I cant find a short reproducable example here
I suppose, that this behavior is related with converting interface to stuct?...
however, [here is a branch of a project, where this bug occures](https://github.com/gucio321/hellspawner/tree/linter-bug)

</details>

**Comment by ldez** [comment]
Hello,

can you provide (as asked in the issue template) the output of your command? Thank you.

**Comment by gucio321** [comment]
yes, this is a duplicate of #1517, I didn't find it, sorry
